sarathi/model_executor/layers/sampler.py

```diff
@@ -219,3 +219,18 @@ def _sample(
             outputs[seq_idx] = SamplerOutput(seq_id, sample_result)
 
     return outputs
+
+
+def _top_k_top_p_with_flashinfer(
+    logits: torch.Tensor, top_ks: torch.Tensor, top_ps: torch.Tensor
+):
+    max_top_k_round = 32
```

Comment:
why 32?

Reply:
It's the same value used in the vllm codebase. No explanation was provided there either so not sure

Reply:
i see, i guess they really don't want it to fail, in any case, can you please move this constant to the top of the file in uppercase

Reply:
Ok will do that. Also realized 32 is reccomend by flashinfer documentation: https://docs.flashinfer.ai/generated/flashinfer.sampling.top_k_top_p_sampling_from_logits.html

Reply:
Hi @AgrawalAmey @kasohrab , thanks for integrating this, I hope the sampling kernel would benefit your project!

The reason it's set to 32 is that we implement a rejection sampling algorithm where each round expect to filter out around half of the choices, and hopefully 32 rounds could find the 1/(2^32) choice (I hope the vocabulary size would not exceed this value). In most cases, the algorithm stops much earlier. But 32 is a safe choice.

Reply:
@yzh119 thanks a lot for sharing the the explanation. The project has greatly benefited from the availability of  flashinfer, thanks again!